### ISSUE TITLE
🐛 Timeout and fail while requesting analyzeDependencies

### ISSUE THREAD

**Issue by andreujuanc** [rome/tools#1247]
#### Docker image: 
`mcr.microsoft.com/vscode/devcontainers/javascript-node:0-14-buster`

#### Rome rage:
```
Environment Variables
  Object {
    COLORTERM: "truecolor"
    LANG: "en_US.UTF-8"
    TERM: "xterm-256color"
    TERM_PROGRAM: "vscode"
  }

User Config
  unset

Rome Version
  10.0.4-beta

Node Version
  14.15.0

Platform
  linux x64 4.19.84-microsoft-standard

Terminal Features
  Object {
    background: "unknown"
    colorDepth: 24
    columns: 185
    cursor: true
    hyperlinks: true
    isTTY: true
    unicode: true
  }

Client Flags
  Object {
    clientName: "cli"
    cwd: "/workspace"
    realCwd: "/workspace"
    silent: false
  }

Server Status
  Object {
    projects: Array [
      Object {id: 0}
      Object {id: 1}
    ]
    server: Object {
      heapTotal: 131_198_976
      pid: 2_655
      uptime: 2.8186217
    }
    workers: Array [
      Object {
        astCacheSize: 0
        heapTotal: 131_198_976
        ownedBytes: 0
        ownedFileCount: 0
        pid: 2_655
        uptime: 2.8184182
      }
    ]
  }

```

## Steps To Reproduce

1. package.json
```
"dependencies": {
    "@attire/core": "^1.0.39",
    "@aws/dynamodb-auto-marshaller": "^0.7.1",
    "aws-sdk": "2.795.0",
    "uuid": "^3.4.0"
  },
  "devDependencies": {
    "@pulumi/aws": "^3.20.0",
    "@pulumi/awsx": "^0.22.0",
    "@pulumi/pulumi": "^2.15.6",
    "@types/jest": "^26.0.19",
    "@types/node": "^10.17.49",
    "@types/uuid": "^8.3.0",
    "jest": "26.*",
    "ts-jest": "26.*",
    "ts-node": "^9.0.0",
    "typescript": "^4.1.3"
  }
```
2. npx rome check

## Expected Results

I was just checking the project out, but it didn't work.

This is the rome check output:

```
 Analyzing files: node_modules/glob/common.js                                                                                                                 elapsed 16s 147 op/s 2_372⚠ Analyzing files: node_modules/@pulumi/pulumi/node_modules/typescript/lib/typescript.js                                                                       elapsed 16s 143 op/s 2_387
⚠ Worker 4 has not responded for 16s seconds. It is unlikely to become responsive. Currently processing:
- analyzeDependencies x 1
  - Object {
      parseOptions: Object {}
      ref: Object {
        manifest: 607
        project: 2
        real: "/workspace/node_modules/@pulumi/pulumi/node_modules/typescript/lib/typescript.js"
        relative: "node_modules/@pulumi/pulumi/node_modules/typescript/lib/typescript.js"
        remote: false
        uid: "profile-service/@pulumi/pulumi/typescript/lib/typescript.js"
      }
    }
ℹ Please open an issue with the details provided above if necessary

 worker 4 project-rome/@internal/core/server/WorkerManager.ts:288:6 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Did not respond for 16666ms and was checked 12 times

  1. <anonymous> (project-rome/@internal/core/server/WorkerManager.ts:288:6)
  2. Timeout.<anonymous> (project-rome/@internal/events/Bridge.ts:178:7)

  ⚠ Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.


 project-rome/@internal/events/Bridge.ts:351:20 internalError/request ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Connection died

  1. ___R$project$rome$$internal$core$common$bridges$WorkerBridge_ts$default.end (project-rome/@internal/events/Bridge.ts:351:20)
  2. ___R$project$rome$$internal$core$server$WorkerManager_ts$default.end (project-rome/@internal/core/server/WorkerManager.ts:106:11)
  3. ___R$project$rome$$internal$core$server$Server_ts$default.onFatalError (project-rome/@internal/core/server/Server.ts:335:21)
  4. onFatalError (project-rome/@internal/core/server/WorkerManager.ts:287:17)
  5. Timeout.<anonymous> (project-rome/@internal/events/Bridge.ts:178:7)

  ℹ Error occurred while requesting analyzeDependencies for node_modules/@pulumi/pulumi/node_modules/typescript/lib/typescript.js

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

✖ Found 1 problem
```

**Comment by andreujuanc** [comment]
UPDATE:
I did steps suggested here https://github.com/rome/tools/issues/1182 but nothing 😢

**Comment by ematipico** [comment]
Any chance you could share the project as repository?

**Comment by andreujuanc** [comment]
@ematipico  no, i cannot, but I will try to setup a repo to repro it. 
Anything I could do meanwhile ? Any other info/data you would need?

**Comment by ematipico** [comment]
I think I have a pretty good idea, Rome is dying while analyzing `typescript.js`, which is a 9MB  file. It's not the first issue we have about the analysis of big files. Not sure how we should tackle the issue, probably @sebmck has some insight.

**Comment by andreujuanc** [comment]
Hi @ematipico 
I setted up repro steps and I even got a slightly different error:
```
worker 1 project-rome/@internal/core/server/WorkerManager.ts:288:6 internalError/fatal  INTERNAL   FATAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Did not respond for 15283ms and was checked 11 times

  1. <anonymous> (project-rome/@internal/core/server/WorkerManager.ts:288:6)
  2. Timeout.<anonymous> (project-rome/@internal/events/Bridge.ts:178:7)

  ⚠ Rome exited as this error could not be handled and resulted in a fatal error. Please report if necessary.


 project-rome/@internal/events/Bridge.ts:351:20 internalError/request ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Connection died

  1. ___R$project$rome$$internal$core$common$bridges$WorkerBridge_ts$default.end (project-rome/@internal/events/Bridge.ts:351:20)
  2. ___R$project$rome$$internal$core$server$WorkerManager_ts$default.end (project-rome/@internal/core/server/WorkerManager.ts:106:11)
  3. ___R$project$rome$$internal$core$server$Server_ts$default.onFatalError (project-rome/@internal/core/server/Server.ts:335:21)
  4. onFatalError (project-rome/@internal/core/server/WorkerManager.ts:287:17)
  5. Timeout.<anonymous> (project-rome/@internal/events/Bridge.ts:178:7)

  ℹ Error occurred while requesting analyzeDependencies for node_modules/aws-sdk/apis/rds-2013-09-09.min.json
```

## REPRO

My goal was to add rome to a side project I'm working on, so it goes like this:
## With VSCode 

1. Install extension ms-vscode-remote.vscode-remote-extensionpack
2. Do this:
```bash
mkdir repro && cd repro
npm init @lambada
code . # If y
```
3. Extension will prompt to "Reopen in Container". Click that button.
4. Add rome (cos it's not yet in the template:
```bash
npx rome init
npx rome check
```

Pulumi and AWS-SDK are the main problem here, but it might be a mixture of both of them as I haven't tried it separately.

**Comment by jamiebuilds** [comment]
Hi! Thank you for opening this issue

We recently made the decision to [migrate the Rome codebase to Rust](https://rome.tools/blog/2021/09/21/rome-will-be-rewritten-in-rust) and making some significant changes to its architecture. We've also decided not to release the previous version written in TypeScript.

We're going through and closing pull requests and issues that were specific to the TypeScript codebase, including this one. Thank you for taking the time to open it.

Thanks again!
